engine/src/execution_graph/logic_controllers/LogicPrimitives.h

```diff
@@ -57,6 +59,8 @@ class BlazingTableView {
 	cudf::column_view const & column(cudf::size_type column_index) const { return table.column(column_index); }
 
 	std::vector<std::string> names() const;
+	// set columnNames
+	void setNames(const std::vector<std::string> &names) { this->columnNames = names; }
```

Comment:
Views are immutable in terms of what columns they have you should never call set names. If you are this is an anti pattern

Reply:
When we do a load_data, we need to be able to set or change the names of the columns after we have loaded them. This function was added in order to be able to do that. There is no way to change the names of a BlazingTable. The alternative is to pass in aliases into load_data, which would mean we would also have to add it to parse, which would mean that this alias handling code would have to be added to 14 files

Reply:
I agree that BlazingTable should be immutable, but I really dont like the idea of having to deal with so much to make it work. Also, its the data that is immutable, i dont think we need to be so strict with the names. I dont think that function would get misused

Reply:
im ok with it for now but I think this will need revisiting